### PR TITLE
Makeup vending machine

### PR DESCRIPTION
**Pull request by CodenameB**
This PR adds in a new vending machine; the younger sister to the barber vending machine; a cosmetics vendor.

This new vending machine can be ordered in pack form from cargo and inserted into any empty vending machine.

Currently, even though makeup barely changes someone's sprite, it is notoriously difficult for anyone to get their hands on the lipstick colors they want to have, since all current sources of it are either always red, or random, and when ordered from cargo come with a bunch of stuff you don't want. It never made much sense that something so simple was a pain to get. This PR seeks to correct this situation and make it easier to get makeup.

In order to further justify its existence, this update also adds eyeshadow as a form of makeup, (in three colors: black, purple, and jade) plus a new shade of lipstick (blue).

The vendor also comes with pocket mirrors, and as the reverse of the barber vendor which has lipstick if you hack it, the cosmetics vendor has hair dye if you hack it.

Sprites:

image

image image

:cl: 
* rscadd: New vending machine: Cosmetics vendor.
* rscadd: Cosmetics vendor pack can be ordered from cargo.
* rscadd: New color of lipstick: Blue
* rscadd: New makeup: Eyeshadow (Black, jade, or purple)
* rscadd: Eyeshadow can be applied just like lipstick, and wiped off the same way, only they use the eye slot instead of the mouth slot.

### CODE REVIEW

**Comment by Exxion** [comment]
I hope you're prepared for memes when people wake up

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -2464,4 +2464,28 @@ var/global/num_vending_terminals = 1
...
 		)
-	pack = /obj/structure/vendomatpack/barbervend
\ No newline at end of file
+	pack = /obj/structure/vendomatpack/barbervend
+
+
+/obj/machinery/vending/makeup
+	name = "Sapphire Cosmetics"
+	desc = "A vending machine full of cosmetics and beauty products."
+	product_slogans = " There is no such thing as natural beauty.; Wear the look of the future.;Be the beauty in the eye of every beholder."
```
Comment (Shadowmech88) [inline code/game/machinery/vending.dm]: There's an extra space at the beginning here.

File: code/game/objects/items/weapons/cosmetics.dm [hunk 2]
```diff
@@ -90,6 +95,81 @@
...
+
+/obj/item/weapon/eyeshadow/random
+	name = "eyeshadow"
+
+/obj/item/weapon/eyeshadow/random/New()
+	colour = pick("purple","jade","black")
+	name = "[colour] eyeshadow"
+	..()
+
+/obj/item/weapon/eyeshadow/attack(mob/M as mob, mob/user as mob)
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cosmetics.dm]: Remove `as mob`.

**Review by Shadowmech88 — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Recommending to add the Flower Hair Pin as a coin-bought premium item

**Comment by yclatious** [comment]
I cant *wait* to be forced lipstick by waifus/trees/clowns every single shift

**Comment by Kammerjunk** [comment]
It's not like the waifu crate never got ordered.
Why, I remember this one black sec officer kept ordering it for the flower pin, opting to leave the sec helmet behind
Why, I remember sec spent that one round applying lipstick to each other instead of helping the fucked-up station
Waifus, everyone

**Comment by Gringonius** [comment]
Why, God?

**Comment by ghost** [comment]
@yclatious  You can already order lipstick crates from cargo and the clown starts with some.  Not to mention it is a guaranteed maintenance spawn on Box Station.

**Comment by yclatious** [comment]
Trust me, I know about Box station maint loot and I also know about cargo lipstick but id rather not get other ways to get general waifushit ingame.

**Comment by Kurfursten** [comment]
The spritework isn't as stand-out amazing as the barber vendor, but I like any new stuff to do as QM - extra fluff vendors are great 👍

**Comment by ghost** [comment]
there is literally nothing wrong with this

**Comment by ghost** [comment]
Codenameb for collab 2017

Truly the hero we need

**Comment by SonixApache** [comment]
Neat.

**Comment by CodenameB** [comment]
@yclatious It is very easy and fast to remove makeup. Hair dye is a much worse fate.

**Comment by ghost** [comment]
beep beep waifucreep

**Comment by Arthurdentist** [comment]
I'm not really sure we need such a vendor now that the barber vendor exists, and I think lipstick being kinda 'rare' was neato. A better PR might have been to add each type of lipstick to the barber vendor, but I opted against that at the time because again, always considered it somewhat 'rare'.

**Comment by ihadtoregisterforthis** [comment]
The blurry part on the sprite is causing me eye damage

**Comment by Riekland** [comment]
There's TECHNICALLY nothing wrong with this
The sprite for the vending machine is far too busy and the blue lipstick is the wrong shade, it needs to be darker or lighter
other than that it's alright I guess but I would ask that if someone spends time ordering it every round that they be told to maybe not do that

**Comment by salamandooo** [comment]
people have the right to spend their creds on whatever they'd like. some people take pride in their apperance

**Comment by ghost** [comment]
Maybe I'm not getting this but why is this a vendor instead of just another supply pack or something

**Comment by CodenameB** [comment]
@Arthurdentist But the problem is, lipstick ISN'T rare, it has a 100% chance to spawn in several spots. It's that the specific color you want (other than red) is hard to find.

@Riekland Do you have a specific shade of blue you prefer? I can darken it.

@Unknown-as-Captain Because again, as I said in my OP, people often want exactly one thing for makeup. They don't want to have to order a bunch of crap they don't want like dresses. As a vendor, they can build the machine, and then everyone can use it to get the one item they want, instead of it just getting dumped down disposals when the person gets the single tube they wanted.

**Comment by ghost** [comment]
image
coral blue 5 is clearly the best color

**Comment by yclatious** [comment]
Pride in their appearance my ass you shlomo, its a bunch of vaguely feminine pixels in a atmos simulation game, played by grown ass man who want to pretend being girls for attention.

Its nothing but literal waifu pandering and waifu fuel, why would anyone want to encourage this sort of shit, does nobody remember the cringy garbage that was flavor text? Sec officers that decided to ignore helmets for the sake of looking cool with berets and hair pins? Does anybody actually like having a bunch of players that focus on one of the worst possible aspects of ss13, its graphics, just for the sake of pretending to be a girl and above liking them, wants to condone that shit?

Im not asking for assistants only, toolboxes, box station, Im just asking that working on a space station isn't a trip to a beauty salon, both for guys and girls to not go and start parading around with dresses and hairpins, special hand crafted hairstyles when the game isn't even slightly focused on personal graphics but on its mechanics, its social interaction and crazy scenarios, this just seems like the biggest load of bullshit to ever try and creep in tbh. Maybe Im the insane one here but if we really wanted to provide extensive character creation choices and such, wouldn't we be going for a bay styled server and higher RP level instead of trying to mostly keep it minimal but not inexistent like we currently do?

**Comment by SonixApache** [comment]
@yclatious code a chemical that makes people that wear makeup turn into vox

**Comment by Kurfursten** [comment]
The idea that you can only be Bay or Nox is just a thread meme. I hate to get academic here but it's a false dichotomy.

I don't do the dress up thing personally, the most 'dress up' I do is wear a botany bandana. But I do enjoy playing service, and I know you're coming for me next. Pretty soon you'll  be saying "this is an atmospherics simulator, this game isn't even slightly focused on food mechanics and if we really wanted to provide a lot of cooking options, wouldn't we be going for a cooking mama/farmville styled server with a station oriented around growing crops??" Don't act like this isn't your endgame, I've seen calls for removing hunger for being 'bay tier'.

Space station 13 is a lot of things for different people where they all interact. Some people like growing plants and cooking, some people like engineering autism projects, some people like chasing security valids, and some people enjoy sitting in the bar roleplaying. It's the intersection of all of that which makes SS13 totally unique from other games.

**Comment by yclatious** [comment]
I said for it to not be a trip to the beauty salon, not that minor shit like bandanas is the end of the world, no, I don't want hunger gone but I don't want a absolute dress up either, I want a balance, the thing I love about /vg/ its its balance and id like for it to keep itself on that balance.

I don't have any endgame of casualizing the server further or making it validtown, that'd be the worst thing I could ever want to do here, but I also don't want it to have waifus parading around in makeup, I didn't think that wanting the balance of neither of these was unthinkable or something so alien that you imediately try and say im going for service next when I actually try my damn best to interact with barkeeps and chefs in every shift I see them in, the last thing id ever do is make service even more useless as its currently mostly just a rp thing, one that I personally go and interact with as often as I can.

Indeed, the combination of all of those is what makes ss13 so great and the balance of /vg is what made me fall in love with this place, and for that id like for that balance to be preserved kurf. Sue me for not wanting to go more hardcore rp or lower rp when I fell in love with the only balanced server out there.

**Comment by Kurfursten** [comment]
Maybe it was a little unnecessarily aggressive of me to accuse you of that, sorry. I know there are some people out there who would see the more roleplay-oriented jobs like Service gutted, though.

**Comment by CodenameB** [comment]
@yclatious "trips to the beauty salon" "wearing hair pins instead of helmets" and "putting lipstick on each other" are all things already in the game. I'm merely making it easier and faster to do so, which means people get back to their jobs faster.

Which would you rather have: People spending 20 minutes at the start of their shift badgering cargo to order female formal wear crates over and over until they get the one lipstick they want and badgering the clown until it lets them into the theater and buys them a hair pin, or, a 2 minute cargo trip to set up a new vending machine? Hell, we could throw it on the map on Bagel in an empty vendor and save even more time.

I offer that my PRs are the balanced choices. After all, we could instead have Bay features where you write custom flavor text for every limb, every cyborg module, and you have to specifically choose the correct military branch and rank to apply for your job role.

**Comment by yclatious** [comment]
The "trips to the beaty salon" exist but are thankfully rare, same with absolutely shit tier hippie players that don't speak at all and just validhunt, relatively rare, mostly due to the latter getting banned quickly so I dont get to experience a lot of rounds with them. Cant tell what's worse honestly, they are such horrid garbage id rather not even care which is worse, I just want them gone or at the bare minimum, if not gone, Id like it reduced to the minimum possible. This goes against the minimum possible I want.

I would rather not have people order this shit at all. Id rather it not exist. Id rather this PR be closed down and for waifushit to be dead and buried and for bagel to have almost any modification rather than more waifu enabling.

A balanced choice of "more waifushit" is not balanced in any way, shape or form, same as a "security starts with a egun" is not balanced either, you want actual balance, you keep it rare instead of making it easier and more common.

**Comment by CodenameB** [comment]
Sorry, but "get rid of it" isn't the compromise solution.

**Comment by yclatious** [comment]
I never "said get rid of it" is the compromise solution, I said that is what I would actually like. The real compromise solution is just for this not to pass and keep the balance as is.

**Comment by CodenameB** [comment]
But we are already at the very low end of content, so that is also not a fair compromise. Leaving things as they are, people who want "less waifu shit" get nearly everything they want, and people who want more RP and character customization, get nothing they want.

**Comment by yclatious** [comment]
No, people who want "less waifu shit" do not get nearly everything they want, if they are validhunting retards they also get the boot, waifu shit gets the same treatment and its by pandering to neither pole that the right balance is obtained. We don't have any validhunting focused changes, we don't have any waifushit focused changes and lo and behold, we dont become either pole of shit nobody wants to get into.

Both poles are at the very low end of content and thats the way they should be. You want to rp, go ahead with the things we got now, you want to validhunt, go with the things you got now, we aren't catered to either polarity exclusively, we do both and by not going to either side we manage to neither be lolkekrofl powergamers or having to fake not knowing the game, having to write a essay so you can play dionae and absolute attention whoring idiots who play as girls. We already got the toned down versions of both of these, don't encourage either damn side.

**Comment by CodenameB** [comment]
Let's not even menton how this doesnt affect you in any way, and you are merely depriving others. See, this isn't about valid hunting vs waifus, those are two are completely unrelated things. This is about people who want more content vs people like you who want less. This PR is well within the middle ground of Hippie style 0 RP and Bay style extreme RP. 

Everyone benefits from this change, because people who want less RP and content (which you call waifu shit) get people back to their jobs and actual work faster, and people who like new content also benefit.

If people changing two pixels on their sprite triggers you so hard, I suggest you make a PR to remove eye color.

**Comment by ghost** [comment]
>people being this mad about waifus
image
makes me want to port more cosmetics

**Comment by yclatious** [comment]
Except the point isn't that its "in the middle" which we already know it isn't, its that it shifts it to a polarity off the middle, which we have already established is not something we should want.

Except no, not everyone benefits it, only a select few will, pandering to the waifufags, a polarity, work efficiency will see no sort of upgrade, at most a downgrade from the kind of players that want this rushing it roundstart on places like bagel for their 4 pixels worth of change, since they seem to think that to roleplay you need either fancy clothes or flavour text anyhow when roleplaying never even needed those, it just needed you to play a character and act like one. At most, its a supplement, but a uneeded one that shifts the balance, and excusing it on bollocks like "waifu players work quicker when they got their items" is bollocks and you know it.

Its garbage. Call me triggered all you want but Im not going to remove basic polarity features just because I dislike your attempt at shifting it to your preferred pole, that'd be contradicting all Ive said so far.

**Comment by Exxion** [comment]
This would have 0 +/- 0 effect on anyone who doesn't use it
The worst thing that could happen is cargo wastes some of their nigh-unlimited cash on it

**Comment by FudgePucker** [comment]
I think that this will benifit clowns because of slip then makeup pranks

**Pull request merged**